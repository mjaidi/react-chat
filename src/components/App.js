import React, { Component } from "react";
import logo from "assets/logo.svg";

class App extends Component {
  goTo(route) {
    this.props.history.replace(`/${route}`);
  }

  login() {
    this.props.auth.login();
  }

  logout() {
    this.props.auth.logout();
  }

  componentDidMount() {
    if (this.props.auth.isAuthenticated()) {
      this.props.history.push("/home");
    }
  }
  render() {
    const { isAuthenticated } = this.props.auth;

    return (
      <div className="app">
        <header className="navbar">
          <div className="navbar-left">
            <img src={logo} className="navbar-logo" alt="logo" />
            <p className="navbar-link" onClick={this.goTo.bind(this, "home")}>
              Home
            </p>
          </div>

          {!isAuthenticated() && (
            <p className="navbar-link" onClick={this.login.bind(this)}>
              Log In
            </p>
          )}
          {isAuthenticated() && (
            <p className="navbar-link" onClick={this.logout.bind(this)}>
              Log Out
            </p>
          )}
        </header>
      </div>
    );
  }
}

export default App;
