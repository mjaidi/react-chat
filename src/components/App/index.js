import React, { Component } from "react";
import logo from "assets/logo.svg";
import "stylesheets/App.scss";

class App extends Component {
  goTo(route) {
    this.props.history.replace(`/${route}`);
  }

  login() {
    this.props.auth.login();
  }

  logout() {
    this.props.auth.logout();
  }

  componentDidMount() {
    if (this.props.auth.isAuthenticated()) {
      this.props.history.push("/home");
    }
  }
  render() {
    const { isAuthenticated } = this.props.auth;

    return (
      <div className="App">
        <button onClick={this.goTo.bind(this, "home")}>Home</button>
        {!isAuthenticated() && (
          <button onClick={this.login.bind(this)}>Log In</button>
        )}
        {isAuthenticated() && (
          <button onClick={this.logout.bind(this)}>Log Out</button>
        )}
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <p>
            Edit <code>src/App.js</code> and save to reload.
          </p>
          <a
            className="App-link"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer"
          >
            Learn React
          </a>
        </header>
      </div>
    );
  }
}

export default App;
