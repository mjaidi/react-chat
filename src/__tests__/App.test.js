import React from "react";
import { shallow } from "enzyme";
import App from "components/App";
import Auth from "../auth/auth0.js";

describe("App", () => {
  let component;
  beforeEach(() => {
    component = shallow(<App auth={new Auth()} />);
    component.instance().login = jest.fn();
    component.instance().logout = jest.fn();
    component.instance().forceUpdate();
  });
  it("should render correctly without crashing", () => {
    expect(component).toMatchSnapshot();
  });

  it("should call login on login button pressed", () => {
    component.find("#login").simulate("click");
    expect(component.instance().login).toHaveBeenCalled();
  });

  it("should call logout on logout button pressed", () => {
    component.setProps({ auth: { isAuthenticated: () => true } });
    component.find("#logout").simulate("click");
    expect(component.instance().logout).toHaveBeenCalled();
  });
});
