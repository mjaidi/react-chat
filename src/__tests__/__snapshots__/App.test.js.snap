// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly without crashing 1`] = `
<div
  className="app"
>
  <header
    className="navbar"
  >
    <div
      className="navbar-left"
    >
      <img
        alt="logo"
        className="navbar-logo"
        src="logo.svg"
      />
      <p
        className="navbar-link"
        onClick={[Function]}
      >
        Home
      </p>
    </div>
    <p
      className="navbar-link"
      id="login"
      onClick={[Function]}
    >
      Log In
    </p>
  </header>
</div>
`;
